Course/Course/Presentation/Dates/CourseDatesViewModel.swift

```diff
@@ -112,18 +112,14 @@ public class CourseDatesViewModel: ObservableObject {
             await getCourseStructure(courseID: courseID)
             if courseDates?.courseDateBlocks == nil {
                 isShowProgress = false
-                errorMessage = CoreLocalization.Error.unknownError
+                courseDates = nil
                 return
             }
             isShowProgress = false
             addCourseEventsIfNecessary()
-        } catch let error {
+        } catch {
             isShowProgress = false
-            if error.isInternetError || error is NoCachedDataError {
-                errorMessage = CoreLocalization.Error.slowOrNoInternetConnection
-            } else {
-                errorMessage = CoreLocalization.Error.unknownError
-            }
+            courseDates = nil
```

Comment:
I guess we don't need to set it nil explicitly.

Reply:
Same [here](https://github.com/edx/edx-mobile-marketplace-ios/pull/28#discussion_r1686203011)